unstructured/partition/email.py

```diff
@@ -82,40 +83,59 @@ def _parse_received_data(data: str) -> list[Element]:
     return elements
 
 
-def _parse_email_address(data: str) -> tuple[str, str]:
-    email_address = extract_email_address(data)
+def _strip_angle_brackets(data: str) -> str:
+    """Remove angle brackets from the beginning and end of the string if they exist.
 
-    PATTERN = r"<[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+>"
-    name = re.split(PATTERN, data.lower())[0].title().strip()
+    Returns:
+    str: The string with surrounding angle brackets removed.
 
-    return name, email_address[0]
+    Example:
+    >>> _strip_angle_brackets("<example>")
+    'example'
+    >>> _strip_angle_brackets("<another>test>")
+    'another>test'
+    >>> _strip_angle_brackets("<<edge>>")
+    '<edge>'
+    """
+    return re.sub(r"^<|>$", "", data)
+
+
+def partition_email_header(msg: EmailMessage) -> list[Element]:
 
+    def append_address_header_elements(header: AddressHeader, element_type: Type[Element]):
```

Comment:
Some context for changing the `msg` type to `EmailMessage`:
"The central class in the [email](https://docs.python.org/3/library/email.html#module-email) package is the [EmailMessage](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage) class, imported from the [email.message](https://docs.python.org/3/library/email.message.html#module-email.message) module. It is the base class for the [email](https://docs.python.org/3/library/email.html#module-email) object model. [EmailMessage](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage) provides the core functionality for setting and querying header fields, for accessing message bodies, and for creating or modifying structured messages."

"There is a default policy used by all classes in the email package. For all of the [`parser`](https://docs.python.org/3/library/email.parser.html#module-email.parser "email.parser: Parse flat text email messages to produce a message object structure.") classes and the related convenience functions, and for the [`Message`](https://docs.python.org/3/library/email.compat32-message.html#email.message.Message "email.message.Message") class, this is the [`Compat32`](https://docs.python.org/3/library/email.policy.html#email.policy.Compat32 "email.policy.Compat32") policy, via its corresponding pre-defined instance [`compat32`](https://docs.python.org/3/library/email.policy.html#email.policy.compat32 "email.policy.compat32"). This policy provides for complete backward compatibility (in some cases, including bug compatibility) with the pre-Python3.3 version of the email package. This default value for the _policy_ keyword to [`EmailMessage`](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage "email.message.EmailMessage") is the [`EmailPolicy`](https://docs.python.org/3/library/email.policy.html#email.policy.EmailPolicy "email.policy.EmailPolicy") policy, via its pre-defined instance [`default`](https://docs.python.org/3/library/email.policy.html#email.policy.default "email.policy.default")."

"The default value for the _policy_ keyword for the [`email.parser`](https://docs.python.org/3/library/email.parser.html#module-email.parser "email.parser: Parse flat text email messages to produce a message object structure.") classes and the parser convenience functions **will be changing** in a future version of Python. Therefore you should **always specify explicitly which policy you want to use** when calling any of the classes and functions described in the [`parser`](https://docs.python.org/3/library/email.parser.html#module-email.parser "email.parser: Parse flat text email messages to produce a message object structure.") module."